src/metabase/models/database.clj

```diff
@@ -254,22 +264,20 @@
             driver.u/default-sensitive-fields))
       driver.u/default-sensitive-fields))
 
-;; when encoding a Database as JSON remove the `details` and `settings` for any User without write perms for the DB.
-;; Users with write perms can see the `details` but remove anything resembling a password. No one gets to see this in
-;; an API response!
-(add-encoder
- #_{:clj-kondo/ignore [:unresolved-symbol]}
- DatabaseInstance
- (fn [db json-generator]
-   (encode-map
-    (if (not (mi/can-write? db))
-      (dissoc db :details :settings)
-      (update db :details (fn [details]
-                            (reduce
-                             #(m/update-existing %1 %2 (constantly protected-password))
-                             details
-                             (sensitive-fields-for-db db)))))
-    json-generator)))
+(methodical/defmethod mi/to-json Database
+  "When encoding a Database as JSON remove the `details` and `settings` for any User without write perms for the DB.
+  Users with write perms can see the `details` but remove anything resembling a password. No one gets to see this in
+  an API response!"
+  [db json-generator]
+  (next-method
```

Comment:
Flexing on people with `next-method` 💪 https://github.com/camsaul/methodical#calling-the-next-most-specific-method-with-next-method

No more `((get-method my-multimethod :i-have-to-know-what-the-next-method-is) x y z)`. Just `(next-method x y z)`